### DISCUSSION TITLE
关于CURL的相关介绍

### DISCUSSION THREAD

**Discussion by pierce-shelby** [liusy58/WhyThis#34] [Ideas]
考虑到工作实践中得到的经验，开发、调试、问题定位都会经常使用到curl命令，想在此处做个简单介绍。

**Comment by pierce-shelby** [comment]
### Linux Curl使用
#### 下载单个文件
默认将输出打印到标准输出中(STDOUT)中
```
curl http://www.centos.org
```
通过-o/-O选项保存下载的文件到指定的文件中: -o: 将文件保存为命令行中指定的文件名的文件中 -O: 使用URL中默认的文件名保存文件到本地
```
# 将文件下载到本地并命名为mygettext.html
curl -o mygettext.html http://www.gnu.org/software/gettext/manual/gettext.html
 
# 将文件保存到本地并命名为gettext.html
curl -O http://www.gnu.org/software/gettext/manual/gettext.html
```
同样可以使用转向字符">"对输出进行转向输出

**Comment by pierce-shelby** [comment]
同时获取多个文件
```
curl -O URL1 -O URL2
```
若同时从同一站点下载多个文件时，curl会尝试重用链接(connection)。

**Comment by pierce-shelby** [comment]
诸如此类，如果觉得可以，今天把相关的内容全部编写完成后合入

**Comment by liusy58** [comment]
我认为可以，但是取一个什么样的名字比较好呢？

**Reply by pierce-shelby** [reply]
感觉有点和whythis相悖 先搁置吧
